### PR TITLE
fix: Ramp on/off and minor issues

### PR DESCRIPTION
**Pull request by stackchain**
- Refactored ramp on / off done by the vendors
- Bug fixes

### CODE REVIEW

File: apps/wallet-mobile/src/features/RampOnOff/common/MaybeShowBuyBanner/MaybeShowBuyBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const {resetShowBuyBannerSmall} = useResetShowBuyBannerSmall()
+
+  const isShowSmallBanner = showSmallBanner && isInThresholdToBuy && !hasZeroPt
+
+  const onCloseSmallBanner = () => {
+    resetShowBuyBannerSmall()
+  }
+
+  if (hasZeroPt) return <ShowBuyBannerBig />
+  if (isShowSmallBanner) return <ShowBuyBannerSmall onClose={onCloseSmallBanner} />
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/RampOnOff/common/MaybeShowBuyBanner/MaybeShowBuyBanner.tsx]:
```suggestion
  const handleOnClose = () => {
    resetShowBuyBannerSmall()
  }

  if (hasZeroPt) return <ShowBuyBannerBig />
  if (isShowSmallBanner) return <ShowBuyBannerSmall onClose={handleOnClose} />
```

File: apps/wallet-mobile/src/features/Staking/Governance/common/helpers.ts [hunk 2]
```diff
@@ -47,20 +47,20 @@ export const useIsGovernanceFeatureEnabled = (wallet: YoroiWallet) => {
 }
 
 export const useGovernanceManagerMaker = () => {
-  const wallet = useSelectedWallet()
+  const {networkId, id: walletId} = useSelectedWallet()
 
   const storage = useStorage()
+  const governanceStorage = storage.join(`wallet/${walletId}/staking-governance/`)
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Staking/Governance/common/helpers.ts:53]: @MichalSzorad can you confirm this update?

File: apps/wallet-mobile/src/features/Swap/common/useSwapTx.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export const useSwapTx = (options?: UseMutationOptions<YoroiUnsignedTx, Error, {
   const wallet = useSelectedWallet()
   const mutation = useMutationWithInvalidations({
     mutationFn: (data) => wallet.createUnsignedTx(data.entries, metadata),
-    invalidateQueries: ['useCreateOrder'],
+    invalidateQueries: [['useCreateOrder']],
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/useSwapTx.ts:32]: bug

File: apps/wallet-mobile/src/TxHistory/TxHistoryList/TxHistoryList.tsx [hunk 4]
```diff
@@ -45,26 +30,17 @@ export const TxHistoryList = (props: Props) => {
...
-  const handleExport = () => Alert.alert(strings.soon, strings.soon)
-  const handleSearch = () => Alert.alert(strings.soon, strings.soon)
-
   return (
     <View style={styles.container}>
-      {(features.txHistory.export || features.txHistory.search) && (
-        <ActionsBanner onExport={handleExport} onSearch={handleSearch} />
-      )}
-
       <Spacer height={18} />
```
Comment (stackchain) [inline apps/wallet-mobile/src/TxHistory/TxHistoryList/TxHistoryList.tsx:38]:
```suggestion
```

File: apps/wallet-mobile/src/TxHistory/TxHistoryList/TxHistoryList.tsx [hunk 5]
```diff
@@ -45,26 +30,17 @@ export const TxHistoryList = (props: Props) => {
...
```
Comment (stackchain) [inline apps/wallet-mobile/src/TxHistory/TxHistoryList/TxHistoryList.tsx]:
```suggestion
```

File: apps/wallet-mobile/src/TxHistory/TxHistoryList/TxHistoryList.tsx [hunk 6]
```diff
@@ -1,30 +1,17 @@
...
-import {BalanceAmount} from '@yoroi/types/lib/balance/token'
-import BigNumber from 'bignumber.js'
 import _ from 'lodash'
-import React, {useEffect, useState} from 'react'
+import React from 'react'
 import {useIntl} from 'react-intl'
-import {Alert, Platform, SectionList, SectionListProps, StyleSheet, View} from 'react-native'
+import {Platform, SectionList, SectionListProps, StyleSheet, View} from 'react-native'
 
 import {Spacer, Text} from '../../components'
```
Comment (stackchain) [inline apps/wallet-mobile/src/TxHistory/TxHistoryList/TxHistoryList.tsx:8]:
```suggestion
import {Text} from '../../components'
```

File: apps/wallet-mobile/src/features/RampOnOff/common/MaybeShowBuyBanner/BuyBannerBig.tsx [hunk 7]
```diff
@@ -3,28 +3,26 @@ import * as React from 'react'
...
 
 const DIMENSIONS = Dimensions.get('window')
 
-const BigBanner = () => {
-  const {theme} = useTheme()
+export const BuyBannerBig = () => {
+  const strings = useStrings()
 
+  const {theme} = useTheme()
+  const styles = React.useMemo(() => getStyles({theme: theme}), [theme])
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/RampOnOff/common/MaybeShowBuyBanner/BuyBannerBig.tsx]: `{theme}`

File: apps/wallet-mobile/src/features/Staking/Governance/common/helpers.ts (hunk 2 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/features/Staking/Governance/common/helpers.ts:53]: This is the only place where the storage is created, so any values used here should be correct.

**Review by michalszorad — APPROVED** [review]

File: apps/wallet-mobile/src/features/RampOnOff/common/MaybeShowBuyBanner/BuyBannerBig.tsx (hunk 7 above)
Comment (stackchain) [inline apps/wallet-mobile/src/features/RampOnOff/common/MaybeShowBuyBanner/BuyBannerBig.tsx]:
```suggestion
  const styles = React.useMemo(() => getStyles({theme}), [theme])
```

File: apps/wallet-mobile/src/features/RampOnOff/common/useShowBuyBannerSmall.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+  const query = useQuery({
+    suspense: true,
+    ...options,
+    queryKey: [wallet.id, storageKeyShowBuyBannerSmall],
+    queryFn: async () => {
+      if (!isInThresholdToBuy) return false
+
+      const nextDateInMs = await rampOnOffStorage.getItem(storageKeyShowBuyBannerSmall)
+      const parsedNextDateInMs = parseNumber(nextDateInMs)
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/RampOnOff/common/useShowBuyBannerSmall.tsx:27]: Does the number need to be parsed here? I think .getItem parses it using JSON.parse by default

**Pull request merged**